lib/assertions.ts

```diff
@@ -955,6 +956,57 @@ export default class DOMAssertions {
     }
   }
 
+  /**
+   * Assert that the tagName of the {@link HTMLElement} or an {@link HTMLElement}
+   * matching the `selector` matches the `expected` tagName, using the
+   * [`tagName`](https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName)
+   * property of the {@link HTMLElement}.
+   *
+   * @param {string} expected
+   * @param {string?} message
+   *
+   * @example
+   * // <h1 id="title">
+   * //   Title
+   * // </h1>
+   *
+   * assert.dom('#title').hasTag('h1');
+   */
+  hasTag(tagName: string, message?: string) {
+    let element = this.findTargetElement();
+    let actual;
+    let expected;
+
+    if (!element) return;
+
+    if (typeof tagName !== 'string') {
+      throw new TypeError(`You must pass a string to "hasTag". You passed ${tagName}.`);
+    }
+
+    if (!TAG_NAMES.includes(tagName.toLowerCase())) {
+      throw new Error(
+        `The tagName '${tagName}' is not a valid HTML tag. You must provide a valid HTML tag`
+      );
+    }
```

Comment:
this seems a bit too restrictive to me. at some point new tag names might be introduced or people will use webcomponents that use custom tag names.

Reply:
Ya this was a toss up, and to be fair I’m happy to remove it. If the caller passes an invalid HTML tag the test will fail anyway, so maybe that’s enough?